measure/src/measure.rs

```diff
@@ -181,19 +246,31 @@ mod tests {
             assert!(measure.as_s() >= 0.99f32 && measure.as_s() <= 1.01f32);
             assert!(measure.as_ms() >= 990 && measure.as_ms() <= 1_010);
             assert!(measure.as_us() >= 999_000 && measure.as_us() <= 1_010_000);
+
+            let (_result, measure) = measure_this!("test", |s| sleep(Duration::from_secs(s)), 1);
+            assert!(measure.as_s() >= 0.99f32 && measure.as_s() <= 1.01f32);
+            assert!(measure.as_ms() >= 990 && measure.as_ms() <= 1_010);
+            assert!(measure.as_us() >= 999_000 && measure.as_us() <= 1_010_000);
         }
 
         // Ensure that this() can be called with a simple closure
         {
             let expected = 1;
             let (actual, _measure) = Measure::this(|x| x, expected, "test");
             assert_eq!(actual, expected);
+
+            let expected = 1;
+            let (actual, _measure) = measure_this!("test", |x| x, expected);
+            assert_eq!(actual, expected);
         }
 
         // Ensure that this() can be called with a tuple
         {
             let (result, _measure) = Measure::this(|(x, y)| x + y, (1, 2), "test");
             assert_eq!(result, 1 + 2);
+
+            let (result, _measure) = measure_this!("test", |x, y| x + y, 1, 2);
+            assert_eq!(result, 1 + 2);
         }
 
         // Ensure that this() can be called with a normal function
```

Comment:
Additionally, a test should be added for the macro here. So that one could do:

```rust
let (result, measure) = Measure::this!(my_multiply, 3, 4);
```

or:

```rust
let (result, measure) = Measure::this!(my_multiply(3, 4));
```

Reply:
> @sudipghimire533 Thanks for this PR! Please resolve the issues from CI: https://buildkite.com/solana-labs/solana/builds/65705#867adc4c-e95a-4e61-90df-6a909ab8aa50
> 
> Additionally:
> 
> 1. I don't think that `Measure::this()` should be deprecated. This new macro should coexist with `Measure::this()`.
> 2. Can this be called `Measure::this!()` instead? I'd prefer that over `Measure::measure_this!()` (if possible).
> 3. I apologize if I missed this—can you please add doc tests/examples for measuring a function with zero arguments.
> 4. Can the macro's impl call `Maeasure::this()` instead of reimplementing? I envision the macro putting the args into a tuple and calling `Measure::this()`.

1) That should make CI happy
2) Oops! Scoping don't applied to macro. Do they?
3) Yeah. I kinda forget that
4) Will try this one. Or may be we can do the vice versa?

Looks like bit of work. Can we have a separate branch on solana itself so that other can also help here?